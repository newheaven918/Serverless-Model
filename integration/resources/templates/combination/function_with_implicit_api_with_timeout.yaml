AWSTemplateFormatVersion: '2010-09-09'
Description: A template to test timeout support for implicit APIs.

Resources:
  MyLambdaFunction:
    Type: AWS::Serverless::Function
    Properties:
      Handler: index.handler
      Runtime: nodejs18.x
      MemorySize: 128
      Timeout: 3
      InlineCode: |
        exports.handler = async () => â€˜Hello World!'
      Events:
        ApiEvent:
          Type: Api
          Properties:
            Path: /hello
            Method: get
            TimeoutInMillis: 5000

Metadata:
  SamTransformTest: true
