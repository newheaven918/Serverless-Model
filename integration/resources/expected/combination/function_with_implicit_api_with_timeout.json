[
  {
    "LogicalResourceId": "MyLambdaFunctionApiEventPermissionProd",
    "ResourceType": "AWS::Lambda::Permission"
  },
  {
    "LogicalResourceId": "MyLambdaFunctionRole",
    "ResourceType": "AWS::IAM::Role"
  },
  {
    "LogicalResourceId": "MyLambdaFunction",
    "ResourceType": "AWS::Lambda::Function"
  },
  {
    "LogicalResourceId": "ServerlessRestApiDeployment",
    "ResourceType": "AWS::ApiGateway::Deployment"
  },
  {
    "LogicalResourceId": "ServerlessRestApiProdStage",
    "ResourceType": "AWS::ApiGateway::Stage"
  },
  {
    "LogicalResourceId": "ServerlessRestApi",
    "ResourceType": "AWS::ApiGateway::RestApi"
  }
]
